node/src/components/network/limiter.rs

```diff
@@ -112,15 +113,10 @@ impl Limiter {
     }
 
     pub(super) fn debug_inspect_unspent_allowance(&self) -> Option<i64> {
-        // Disabled temporarily, as it results in a panic:
-
-        // node panicked: Cannot start a runtime from within a runtime. This happens because a
-        // function (like `block_on`) attempted to block the current thread while the thread is
-        // being used to drive asynchronous tasks.
-
-        None
-
-        //Some(self.data.resources.blocking_lock().available)
+        let resources = Arc::clone(&self.data.resources);
+        thread::spawn(move || Some(resources.blocking_lock().available))
```

Comment:
I'm not a fan of this, as it is unclear what the underlying issue is. In other words, if this solves the issue, we should consider rearchitecting the entire thing.

 Let's give this another look before taking this measure. If we can't due to time constraints, we'll merge as is.

Reply:
Further investigation showed that the original implementation was also panicking. We didn't observe the crash though, because on the `feat-1.6` branch the `Limiter` was implemented as a trait, hence, while testing with NCTL we're actually never invoking `blocking_lock()` due to the [default implementation](https://github.com/casper-network/casper-node/blob/feat-1.6/node/src/components/small_network/limiter.rs#L56-L58) (NCTL doesn't use [ClassBaseLimiter](https://github.com/casper-network/casper-node/blob/feat-1.6/node/src/components/small_network/limiter.rs#L105)), which contains the [problematic implementation](https://github.com/casper-network/casper-node/blob/feat-1.6/node/src/components/small_network/limiter.rs#L254-L256).

On the `feat-fast-sync-v2` branch, however, the `Limiter` is a struct, hence we've been always invoking the panicking `blocking_lock()`.

The workaround proposed in this PR is again up for the review, with an explanatory comment added in 2a759d5259dc659573a49c6c6a770ccb190873ef.